uvtools/dspec.py

```diff
@@ -212,63 +364,97 @@ def high_pass_fourier_filter(data, wgts, filter_size, real_delta, clean2d=False,
     # fft back to input space
     if clean2d:
         d_mdl = np.fft.fft2(_d_cl, axes=(0, 1))
+        d_res = np.fft.fft2(_d_res, axes=(0, 1))
     else:
         d_mdl = np.fft.fft(_d_cl)
+        d_res = np.fft.fft(_d_res)
 
-    # get residual in data space
-    d_res = (data - d_mdl) * ~np.isclose(wgts * win, 0.0)
+    if not linear:
+        d_res = (data - d_mdl) * ~np.isclose(wgts * win, 0.0)
 
     return d_mdl, d_res, info
 
-def linear_delay_filter(data, wgts, df, filter_centers, filter_widths, filter_factors, cache = {}):
+def linear_filter(data, wgts, delta_data, filter_dimensions, filter_centers, filter_widths, filter_factors, cache = {}):
     '''Apply a linear delay filter to waterfall data.
+        Due to performance reasons, linear cleaning only supports separable delay-rate filters.
```

Comment:
I think we now use the `delay / fringe-rate` terminology instead of `delay-rate`

Reply:
I've changed this.